cpu/stm32/periph/dma.c

```diff
@@ -26,8 +26,12 @@
 #include "assert.h"
 #include "pm_layered.h"
 
-#if CPU_FAM_STM32F3
-#error "DMA is not supported on STM32F3"
+#if CPU_FAM_STM32F3 && !defined(CPU_LINE_STM32F334x8)
+#warning "DMA support on STM32F3 was only tested on STM32F334r8"
+/* If you try it on another STM32F3 and it doesn't work, you may need to
+ * add the dma_stop() lines in the _transfer_dma() function of the file
+ * cpu/stm32/periph/spi.c what is needed and currently active only for
+ * CPU_LINE_STM32F334x8. */
 #endif
```

Comment:
Since you are adding the missing bits for F3, you can just remove this code. There's no need to limit this to only stm32f334r8 when it might also work for others. The feature will be provided and should be configured at board level anyway.